### ISSUE TITLE
tld validation

### ISSUE THREAD

**Issue by DRTEK** [m2osw/libtld#4]
hello,
when i run this lines of code 
char * uri="advisor-z2-ngprod-1997768525.us-west-2.elb.amazonaws.com"
r = tld(uri_lowercase, &info);

r returns TLD_RESULT_INVALID

**Comment by AlexisWilke** [comment]
Hi DRTEK,

Sorry for the delay. It is likely that a sub-domain was added. The AWS domain names are actually considered at the 3rd level. So it could be that libtld doesn't (yet) know of elb.

**Comment by DRTEK** [comment]
Hello,
no worries ,
can you tell me how we can update the tld database / list in the library ?
thanks

**Comment by AlexisWilke** [comment]
There is a `tld_data.xml` file. The aws definitions would be in the last section (_entrepreneurial_).

The file gets compiled with Qt and then linked in the library as binary. So if you are not compiling the libtld project, you can't update the table. At some point, I'm  thinking I would need to create the compressed table as a binary file I can load on startup... that way we could update the data without recompiling everything.

I'm also working on an update of that file, so we should soon have a new version with the more or less current TLDs (it changes really fast now).

**Comment by AlexisWilke** [comment]
Okay, I published version 1.6.0 with a fix.

The fact is that it wasn't working correctly. The definition was there, but the `'*'` was not properly implemented yet.

I added a test to specifically verify your URL above (and a few others) and I can tell that the newer version will work as expected. The other URLs that accept any sub-domain name as a part of their TLD all work as expected.

Let me know if you have any other problem with 1.6.0.

Thank you for reporting the issue.

See https://github.com/m2osw/libtld/releases/tag/1.6.0
